test/behavior/bugs/3779.zig

```diff
@@ -7,9 +7,9 @@ const ptr_tag_name: [*:0]const u8 = tag_name;
 
 test "@tagName() returns a string literal" {
     if (builtin.zig_backend == .stage1) return error.SkipZigTest; // stage1 gets the type wrong
-    try std.testing.expectEqual(*const [13:0]u8, @TypeOf(tag_name));
-    try std.testing.expectEqualStrings("TestEnumValue", tag_name);
-    try std.testing.expectEqualStrings("TestEnumValue", ptr_tag_name[0..tag_name.len]);
+    try std.testing.expect(*const [13:0]u8 == @TypeOf(tag_name));
```

Comment:
more of a question: whats missing for `expectEqual` and `expectEqualStrings` to work?

Reply:
Good question, and might be a useful note for posterity:

`std.testing.expectEqual*()` functions try to be helpful to the user by printing out, when a difference is found, what the difference was or where inside the string/slice it began. Unfortunately this means that they require `std.debug.print()` to work, which then carries with it a lot more of the stdlib than strictly necessary for equality testing. In the case of the C backend, this means that in order for it to work freestanding it requires `asm()` statements to compile, which are currently not working in stage2. I confirmed with Andrew on Discord chat that it's OK to move behavior tests to a plain `std.testing.expect()` if needed:

![image](https://user-images.githubusercontent.com/357324/159010943-8095743c-a30c-4473-b4db-3e854b4cb4d0.png)


As a side note: printing _does_ already work if we link with `-lc`, which makes `std.os.system == std.c`: https://github.com/ziglang/zig/blob/2c434cddd6624cfd5c71f081b9445936d25c7703/lib/std/os.zig#L64-L65